App Bundle reference (#1560)
giantswarm/docs: src/content/app-platform/app-bundle/index.md

the namespace you have access to, and does not go beyond, then installing it within the Management Cluster
should work.

</div>

#### Configuring target Workload Cluster for the nested App CRs

Once the App CR name and configuration for unique App Operator are taken care of, the only thing left is to
configure the bundle so that nested App CRs are placed into correct Workload Cluster namespace, and with the
correct configuration. It requires two steps.

Firstly, the App CR's `.metadata.namespace` and `.spec.namespace` fields must be equal, meaning must reference
the same namespace. Let's consider the snippet from the previous paragraph and enrich it with these settings.

```yaml
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: bundle-wc1
  namespace: wc1
spec:
  ...
  kubeConfig:
    inCluster: true
  name: bundle-app
  namespace: wc1
  ...
```

Secondly, each app bundle Helm Chart exposes the `clusterName` and `organization` configuration options in its
`values.yaml` file, see below.

```yaml

> clusterName: ""

Reviewer: My only question is: Why did we move to clusterName instead of clusterID ?

Author: I don't remember of when exactly this decision has been taken, but I doubt it was one team behind it as there seem to be a global movement towards cluster names. At least I've seen it being used all over the places.

---
linkTitle: App Bundle
title: App Bundle reference
description: Overview of the the App Bundle concept, how it works, and how to configure it.
menu:
  main:
    parent: app-platform

> weight: 30
> aliases:
>  - /reference/app-bundle/
> last_review_date: 2022-09-14

Reviewer 2: Aliases should only be created if a page is moved, to have a redirect from the old URL (the alias) to the new URL.

```suggestion
weight: 30
last_review_date: 2022-09-14
```

owner:
  - https://github.com/orgs/giantswarm/teams/team-honeybadger
user_questions:
 - What is the App Bundle?

>  - How does the App Bundle installation work?

Reviewer 2: ```suggestion
 - How does installation of an app bundle work?
```

>  - How to properly configure the App Bundle?

Reviewer 2: ```suggestion
 - How to properly configure an app bundle?
```

>  - Are there any caveats in using App Bundle?

Reviewer 2: ```suggestion
 - Are there any caveats in using app bundles?
```

```yaml
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: bundle-wc1
  namespace: wc1
spec:
  ...
  kubeConfig:
    inCluster: true
  name: bundle-app
  namespace: wc1
  ...
```

Secondly, each App Bundle Helm Chart exposes the `clusterName` and `organization` configuration options in its
`values.yaml` file, see below.

```yaml
clusterName: ""
organization: ""
...
```

It is obligatory, if no further customization is needed, to at minimum fill in these two fields and provide such
`values.yaml` as user configuration to the App CR. Let's get back to the example above and enrich it further.

```yaml
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: bundle-wc1
  namespace: wc1
spec:
  ...
  kubeConfig:
    inCluster: true
  name: bundle-app
  namespace: wc1
  userConfig:
    configMap:
      name: bundle-wc1-userconfig
      namespace: wc1
  ...
---
apiVersion: v1
data:
  values: |
    clusterName: wc1
    organization: testorg
kind: ConfigMap
metadata:
  name: bundle-wc1-userconfig
  namespace: wc1
```

> Please refer to the [App Configuration]({{< relref "/app-platform/app-configuration" >}}) to understand how the

Reviewer 2: ```suggestion
Please refer to [App Configuration]({{< relref "/app-platform/app-configuration" >}}) to understand how the
```

> user configuration works, and to the [kubectl gs template app]({{< relref "/ui-api/kubectl-gs/template-app.md" >}}) to get of how you can automate this process.

Reviewer 2: ```suggestion
user configuration works, and to the [kubectl gs template app]({{< relref "/ui-api/kubectl-gs/template-app.md" >}}) reference to understand how you can automate this process.
```
